From 6fac6696adffc7b4f2bd47e12e8cae842736813f Mon Sep 17 00:00:00 2001
From: x <a@b.c>
Date: Wed, 2 Sep 2026 20:23:36 +0000
Subject: [PATCH] Recognise riscv64gc and riscv32gc as valid architectures

rust names the riscv targets riscv64gc-* and riscv32gc-*, and OE passes that
triple straight through to configure. config.sub rejects it.

Upstream-Status: Pending
---
 build/autoconf/config.sub                                     | 2 ++
 .../mozbuild/test/configure/test_toolchain_configure.py       | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/build/autoconf/config.sub b/build/autoconf/config.sub
index 4aaae46f6..10020ca01 100755
--- a/build/autoconf/config.sub
+++ b/build/autoconf/config.sub
@@ -1395,8 +1395,10 @@ case $cpu-$vendor in
 			| pyramid \
 			| riscv \
 			| riscv32 \
+			| riscv32gc \
 			| riscv32be \
 			| riscv64 \
+			| riscv64gc \
 			| riscv64be \
 			| rl78 \
 			| romp \
diff --git a/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py b/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
index 481ce4687..b5ce7785b 100644
--- a/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
+++ b/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
@@ -1328,6 +1328,10 @@ class LinuxCrossCompileToolchainTest(BaseToolchainTest):
         "mips-unknown-linux-gnu": big_endian + {"__mips__": 1},
         "riscv32-unknown-linux-gnu": little_endian + {"__riscv": 1, "__riscv_xlen": 32},
         "riscv64-unknown-linux-gnu": little_endian + {"__riscv": 1, "__riscv_xlen": 64},
+        "riscv32gc-unknown-linux-gnu": little_endian
+        + {"__riscv": 1, "__riscv_xlen": 32},
+        "riscv64gc-unknown-linux-gnu": little_endian
+        + {"__riscv": 1, "__riscv_xlen": 64},
         "sh4-unknown-linux-gnu": little_endian + {"__sh__": 1},
     }
 
